### PR TITLE
Fix Batch Norm compatibility with 3D inputs

### PR DESCRIPTION
**Pull request by mdering**
the theano backend now uses dnn_batch_normalization which only supports
up to 4-dimensional input. This breaks any 5-d layers such as 3D
convolutions.

### CODE REVIEW

File: keras/backend/theano_backend.py [hunk 1]
```diff
@@ -386,7 +386,7 @@ def normalize_batch_in_training(x, gamma, beta,
 def batch_normalization(x, mean, var, beta, gamma, epsilon=0.0001):
     '''Apply batch normalization on x given mean, var, beta and gamma.
     '''
-    if theano.config.device.startswith('cuda') or theano.config.device.startswith('gpu'):
+    if x.ndim < 5 and (theano.config.device.startswith('cuda') or theano.config.device.startswith('gpu')):
```
Comment (fchollet) [inline keras/backend/theano_backend.py]: In that case shouldn't it be `if x.ndim == 4`?

Also this line is becoming too long, I'd suggest creating a `use_cudnn` intermediate variable.

**Comment by mdering** [comment]
well it could potentially be fewer than 4 dimensions, right? Anyway this just mimics the value check in theano, unfortunately this throws a value error, along with a few other errors, so catching the value error and passing is not an option. I can create an intermediate variable, is that the only issue you see? https://github.com/Theano/Theano/blob/e5c41e766abdbd15c96eab6bd7e2b9136659885c/theano/sandbox/cuda/dnn.py#L2764

File: keras/backend/theano_backend.py (hunk 1 above)
Comment (mdering) [inline keras/backend/theano_backend.py]: well it could potentially be fewer than 4 dimensions, right? Anyway this just mimics the value check in theano, unfortunately this throws a value error, along with a few other errors, so catching the value error and passing is not an option. I can create an intermediate variable, is that the only issue you see? https://github.com/Theano/Theano/blob/e5c41e766abdbd15c96eab6bd7e2b9136659885c/theano/sandbox/cuda/dnn.py#L2764

**Comment by fchollet** [comment]
LGTM
